File: app/src/test/java/com/example/androidcodingtest/login/LoginPresenterTest.java
package com.example.androidcodingtest.login;

import com.example.androidcodingtest.models.UserAccount;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.verify;

public class LoginPresenterTest {
    private LoginInteractor.Presenter presenter;
    @Mock
    private LoginInteractor.View view;

    @Before
    public void setUp(){
        MockitoAnnotations.initMocks(this);
        presenter = new LoginPresenter(view);
    }

    @Test
    public void login_InvalidUser_LoginFail(){
        String user = "user#gmail.com";
        String password = "A1!";
        presenter.login(user, password);
        verify(view, Mockito.never()).loginSuccess(anyString(), (UserAccount) Mockito.any());
    }

    @Test
    public void login_InvalidPassword_LoginFail(){
        String user = "user@gmail.com";
        String password = "asadsa";
        presenter.login(user, password);
        verify(view, Mockito.never()).loginSuccess(anyString(), (UserAccount) Mockito.any());
    }

    @Test
    public void login_ValidInformation_LoginSuccess(){
        String user = "user@gmail.com";
        String password = "Ads1@";

        LoginPresenter loginPresenter = new LoginPresenter(view);

        Boolean validUser = loginPresenter.validateUser(user);
        Boolean validPassword = loginPresenter.validatePassword(password);

        Assert.assertEquals(true, validUser);
        Assert.assertEquals(true, validPassword);
    }
}

File: app/src/test/java/com/develop/tcs_bank/login/LoginPresenterTest.kt
package com.develop.tcs_bank.login

import com.develop.tcs_bank.presentation.login.LoginContract
import com.develop.tcs_bank.presentation.login.LoginPresenter
import com.develop.tcs_bank.presentation.login.LoginView
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.*
import org.mockito.MockitoAnnotations

class LoginPresenterTest {


    private lateinit var loginPresenter: LoginContract.Presenter

    @Mock
    private lateinit var loginView: LoginView

    @Before
    fun setUp(){

        MockitoAnnotations.initMocks(this)
        loginPresenter = LoginPresenter(loginView)

    }

    @Test
    fun processLogin_NullString(){

        loginPresenter.processLogin("","")
        verify(loginView, never())

    }
}